src/main/java/org/mockito/internal/junit/JUnitRule.java

```diff
@@ -8,87 +8,75 @@
 import org.junit.runners.model.Statement;
 import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
+import org.mockito.quality.Strictness;
 import org.mockito.internal.util.MockitoLogger;
 import org.mockito.junit.MockitoRule;
 
 /**
  * Internal implementation.
  */
 public class JUnitRule implements MockitoRule {
-	
-	private final MockitoLogger logger;
-    private final boolean silent;
+
+    private final MockitoLogger logger;
+    private final MockitoTestListener listener;
 
     /**
      * @param logger target for the stubbing warnings
-     * @param silent whether the rule
+     * @param strictness how strict mocking / stubbing is concerned
      */
-    public JUnitRule(MockitoLogger logger, boolean silent) {
-		this.logger = logger;
-        this.silent = silent;
+    public JUnitRule(MockitoLogger logger, Strictness strictness) {
+        this.logger = logger;
+        switch (strictness) {
```

Comment:
Since strictness is now public I may not be a good idea anymore, though I liked it.